### PR TITLE
Adds notify-send bug warning when topgrade is run

### LINKED ISSUE #288: Topgrade can't close the process
Hello there! It's been a couple of days that I noticed a strange behavior from Topgrade, never happened before. Basically it executes but won't close, the process hangs up indefinitely. 

## What actually happened?
After updating everything, topgrade won't stop running. Only solved by manually killing the process from htop.

## Additional Details
Operating System: Manjaro Linux
KDE Plasma Version: 5.26.4
KDE Frameworks Version: 5.101.0
Qt Version: 5.15.7
Kernel Version: 5.15.85-1-MANJARO (64-bit)
Graphics Platform: X11
Processors: 12 × AMD Ryzen 5 3600XT 6-Core Processor
Memory: 15,6 GiB of RAM
Graphics Processor: NV137
Topgrade -V 10.2.4

<details>

```
[nicola@nicola-desktop ~]$ sudo topgrade
[sudo] password for nicola: 

―― 02:28:43 - System update ――――――――――――――――――――――――――――――――――――――――――――――――――――
Preparing...
Synchronizing package databases...
Nothing to do.
Transaction successfully finished.

―― 02:28:45 - Configuration update ―――――――――――――――――――――――――――――――――――――――――――――

―― 02:28:45 - pip3 ―――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――
Requirement already satisfied: pip in /root/.local/lib/python3.10/site-packages (22.3.1)
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv

―― 02:28:46 - Flatpak User Packages ――――――――――――――――――――――――――――――――――――――――――――
Looking for updates…

Nothing to do.

―― 02:28:46 - Flatpak System Packages ――――――――――――――――――――――――――――――――――――――――――
Looking for updates…

Nothing to do.

―― 02:28:46 - snap ―――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――
All snaps up to date.

―― 02:28:47 - Firmware upgrades ――――――――――――――――――――――――――――――――――――――――――――――――
Firmware metadata last refresh: 10 hours ago. Use --force to refresh again.
Devices with no available firmware updates: 
 • CT500P5SSD8
 • System Firmware
 • WDC WD10EZEX-08WN4A0
Devices with the latest available firmware version:
 • UEFI dbx
No updates available

―― 02:28:47 - Summary ――――――――――――――――――――――――――――――――――――――――――――――――――――――――――
System update: OK
config-update: OK
pip3: OK
Flatpak: OK
snap: OK
Firmware upgrades: OK
_

```
<pre>

</pre>
</details>

### PR DESCRIPTION
**Pull request by DottoDev**
Closes #288 
## Standards checklist:

- [x] The PR title is descriptive.
- [x] The code compiles (`cargo build`)
- [x] The code passes rustfmt (`cargo fmt`)
- [x] The code passes clippy (`cargo clippy`)
- [x] The code passes tests (`cargo test`)
- [ ] *Optional:* I have tested the code myself
    - [ ] I also tested that Topgrade skips the step where needed

If you developed a feature or a bug fix for someone else and you do not have the
means to test it, please tag this person here.

### CODE REVIEW

**Comment by Focshole** [comment]
There is one typo, preambe instead of preamble (sorry I don't have my laptop rn)

**Comment by DottoDev** [comment]
@Focshole fixed now

**Comment by Focshole** [comment]
Wait, one more thing

File: src/main.rs [hunk 1]
```diff
@@ -86,6 +86,13 @@ fn run() -> Result<()> {
     debug!("Binary path: {:?}", std::env::current_exe());
     debug!("Self Update: {:?}", cfg!(feature = "self-update"));
 
+    if config.display_preamble() {
```
Comment (Focshole) [inline src/main.rs:89]: So add here that check, if possible

Comment (DottoDev) [reply src/main.rs:89]: true, haven't thought about that

Comment (DottoDev) [reply src/main.rs]: Fixes with latest commit

**Comment by Focshole** [comment]
It looks ok to me! Well done 👍

**Comment by luciodaou** [comment]
> Wait, one more thing If --skip-notify flag is specified, this warning should not be shown I think. Do you agree?

I agree. It makes no sense to me to get the warning when using the flag:

image

**Comment by DottoDev** [comment]
@luciodaou i have tried it out and it worked for me. Just to make sure it does show up for you with --skip-notify. Also could you make sure you are using the latest commit(or now also dev branch)

**Comment by luciodaou** [comment]
> @luciodaou i have tried it out and it worked for me. Just to make sure it does show up for you with --skip-notify. Also could you make sure you are using the latest commit(or now also dev branch)


Fedora 37 on WSL, topgrade installed through `cargo`:
image

Arch on WSL, topgrade installed through `cargo`:
image

Windows 11, topgrade installed through `cargo`:
image

**Pull request merged**